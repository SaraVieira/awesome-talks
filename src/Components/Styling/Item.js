import { Link } from 'react-router-dom'
import styled from 'styled-components'
import remcalc from 'remcalc'

export default styled(Link)`
    background-color: #fbfbfb;
    border: ${remcalc(1)} solid #2e3d46;
    padding: ${remcalc(6)} ${remcalc(15)};
    color: #2e3d46;
    border-radius: ${remcalc(5)};
    transition: all 300ms ease-in-out;
    margin: ${remcalc(5)};
    text-decoration: none;

    &:after {
        width: 0;
        height: 0;
    }

    &:hover {
        border: ${remcalc(1)} solid #337294;
        color: #337294;
    }
`
