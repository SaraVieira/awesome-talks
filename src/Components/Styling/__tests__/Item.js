import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import Shallow from '../../../Tests/shallow'
import Item from '../Item'

test('Renders Item', () => {
    const Wrapper = Shallow(
        <MemoryRouter>
            <Item to="https://google.com" />
        </MemoryRouter>
    )
    expect(Wrapper.children()).toMatchSnapshot()
})
