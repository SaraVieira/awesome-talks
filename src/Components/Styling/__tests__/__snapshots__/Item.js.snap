// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Item 1`] = `
<Styled(Link)
  to="https://google.com"
/>
`;
